### DISCUSSION TITLE
User and Group Permission Issues in Tethys

### DISCUSSION THREAD

**Discussion by tmcstraw** [tethysplatform/tethys#737] [Q&A] [answered]
Hey Team,

This may be in the feature request category, but I have spun up a portal for my company and have been working with user permissions to restrict access so that certain departments can only access apps related to their work.  One thing we would like is for the apps that a user doesn't have access to to not show up in the apps library page at all (much like how admin users can see partially installed applications but normal users can not).  This would allow us to spin up apps for multiple departments in the same portal, avoid risk of them editing apps that don't apply to them, and prevent the need for spinning up more than one app portal.  Is there a way to toggle the app visibility in the library based on user permissions? Curious what light the group can shed on this, including any workarounds they may recommend.

***UPDATE****

I figured out the problem, needed to enable restrict app access in the portal config file.  Consider this question resolved.  Thanks

**Comment by swainn** [comment, marked as answer]
Hi @tmcstraw! Sorry for the delay in responding. Sounds like you figured the issue out, but I thought I'd post a few resources for future reference.

You can control access to apps via permission groups. Create a new permission group in the Admin pages of Tethys portal and then use the Apps section to select apps that the group should have access to. Then you can assign users to groups to give them access to those apps. See this documentation to learn more: http://docs.tethysplatform.org/en/stable/tethys_portal/admin_pages.html#assign-app-permission-groups

However, by default, restricting access to apps is **disabled by default**. To enable this feature, set the following setting to `True` in your `portal_config.yaml`:

`ENABLE_RESTRICTED_APP_ACCESS: True`

See the documentation on the Tethys Portal Configuration for a description of this and other settings: http://docs.tethysplatform.org/en/stable/tethys_portal/configuration.html

**Reply by tmcstraw** [reply]
Thanks @swainn! No worries at all, the portal config file was the step I was missing, so it seemed like it was broken because I hadn't enabled that.  Might be helpful to site the second link in the group permissions documentation as well for future users.  Thanks for the help :)

**Comment by swainn** [comment]
Done. See #739

**Reply by tmcstraw** [reply]
Awesome Thanks!
